bitnami/mastodon/4/debian-11/rootfs/opt/bitnami/scripts/libmastodon.sh

```diff
@@ -33,7 +33,7 @@ mastodon_validate() {
         error_code=1
     }
     check_empty_value() {
-        if is_empty_value "${!1}"; then
+        if [[ -v "$1" ]] && is_empty_value "${!1}"; then
```

Comment:
Why did you add it? I don't consider it necessary.

Reply:
So this is to prevent the case that I ran into where the variable wasn't just unset, but was undefined. The line following would be unreachable and it would be very difficult to determine which variable was being tested. Maybe `check_empty_value` is the wrong place for this semantically, but it should be checked somewhere.

I just realized that this check is actually wrong. If the variable is undefined, it will do nothing. It should be:
```suggestion
        if [[ ! -v "$1" ]] || is_empty_value "${!1}"; then
```
This will shortcut to the error we want on the next line if the variable is unset. And if it is set, but empty, it will also go to the error on the next line. And if it's set and not empty nothing will happen.

Reply:
`[[ ! -v "$1" ]]` will be always true. Take into account that all the variables are set (at least to an empty value) in [mastodon-env.sh](https://github.com/bitnami/containers/pull/30395/files#diff-4327d3a68798929b5596005ef5ccb2a1de317d317ba61d4c8bfad83637552375). So it is only necessary to check if empty with `-z`. That's what `is_empty_value` does.

Reply:
That's my point. As it stands, it's not true *now*. It's a sanity check because someone made a typo that is fixed in this PR. It prevents someone from making a mistake that's hard to recover from in the future.

Reply:
Specifically I'm referring to the `MASTODON-S3-ALIAS` unset variable on line 114. Someone wrote the name wrong and it failed poorly because the error only states that the variable is unset, not what the variable is. To debug this I had to go to source and find a variable that didn't make sense. If I had this test I would have known exactly what I was looking for immediately.

Reply:
I see your point. However, this would imply changing a lot of files since `check_empty_value` is used many times for all the solutions. Also, as the name indicates, it is supposed to check if a variable is empty. Another issue is that `-v` flag is not recognized by BASH <= 3, and could produce errors.

Due to this, we prefer not to add this validation and show the error in case of a typo.

Reply:
Thats fair enough, I guess. Don't need to upend the world. I'll remove it.